Completing the final task of Lesson 8 on Interacting with the server

lera-kolesnik opened the pull request.

Commit:
Added task Учебный проект: Корзина, часть 2

lera-kolesnik committed:
Completed fourth task in Lesson 8

(8-4) Implement renderModal, onProductUpdate and onSubmit methods for handling shopping cart functionality. These methods enable the display of shopping cart modal, updating the cart items' quantity and prices, and submitting the order form with appropriate handling of the response

lera-kolesnik committed:
Merge branch 'master' into master

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

lera-kolesnik committed:
fix conflicts

lera-kolesnik committed:
Completed fourth task in Lesson 8

(8-4) Implement renderModal, onProductUpdate and onSubmit methods for handling shopping cart functionality. These methods enable the display of shopping cart modal, updating the cart items' quantity and prices, and submitting the order form with appropriate handling of the response

jsru commented:
Добавляю преподавателя (@koalex) для код-ревью.
